examples/step-63/step-63.cc

```diff
@@ -876,6 +1002,29 @@ namespace Step63
   }
 
 
+  // @sect4{<code>AdvectionProblem::solve</code>}
+
+  // Before we can solve the system, we must first set up the multigrid
+  // preconditioner. This is requires the setup of the transfer between levels,
+  // the coarse matrix solver, and the smoother. This setup follows almost
+  // identically to Step-16, the main difference being the various smoothers
+  // defined above and the fact that we need different interface edge matrices
+  // for in and out since our problem is non-symetric. (In reality, for this
+  // tutorial these interface matrices are empty since we are only using global
+  // refinement, and thus have no refinement edges. However, we have still
+  // included both here since if one made the simple switch to an adaptively
+  // refined method, the program would still run correctly.)
+
+  // The last thing to note is that since our problem is non-symetric, we must
+  // use an appropriate Krylov subspace method. We choose here to
+  // use GMRES since it offers the guarentee of residual reduction in each
+  // iteration. The major disatvantage to GMRES is that, for each iteration, we
+  // must store an additional temporary vector as well as compute an additional
+  // scalar product. However, the goal of this tutorial is to have very low
```

Comment:
I don't know if you want to say that we typically relax the requirement of an additional temporary vector for *each* iteration by using the restarted GMRES iteration where we, as per your setting of "50" auxiliary vectors below, would run 48 iterations before throwing away the vectors from the current state.